cpp/tests/prims/mg_transform_reduce_e.cu

```diff
@@ -0,0 +1,445 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/high_res_clock.h>
+#include <utilities/base_fixture.hpp>
+#include <utilities/device_comm_wrapper.hpp>
+#include <utilities/test_graphs.hpp>
+#include <utilities/test_utilities.hpp>
+#include <utilities/thrust_wrapper.hpp>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/partition_manager.hpp>
+
+#include <cuco/detail/hash_functions.cuh>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/matrix_partition_view.hpp>
+#include <cugraph/prims/copy_to_adj_matrix_row_col.cuh>
+#include <cugraph/prims/transform_reduce_e.cuh>
+
+#include <thrust/count.h>
+#include <raft/comms/comms.hpp>
+#include <raft/comms/mpi_comms.hpp>
+#include <raft/handle.hpp>
+#include <rmm/device_scalar.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <gtest/gtest.h>
+
+#include <random>
+
+template <typename vertex_t, typename... Args>
+struct property_transform : public thrust::unary_function<vertex_t, thrust::tuple<Args...>> {
+  int mod{};
+  property_transform(int mod_count) : mod(mod_count) {}
+  constexpr __device__ auto operator()(const vertex_t& val)
+  {
+    cuco::detail::MurmurHash3_32<vertex_t> hash_func{};
+    auto value = hash_func(val) % mod;
+    return thrust::make_tuple(static_cast<Args>(value)...);
+  }
+};
+template <typename vertex_t, template <typename...> typename Tuple, typename... Args>
+struct property_transform<vertex_t, Tuple<Args...>> : public property_transform<vertex_t, Args...> {
+};
+
+template <typename Tuple, std::size_t... I>
+auto make_iterator_tuple(Tuple& data, std::index_sequence<I...>)
+{
+  return thrust::make_tuple((std::get<I>(data).begin())...);
+}
```

Comment:
This isn't about converting an std tuple to thrust tuple. It is about getting a zipped iterator from an std::tuple of device_vectors. I will move it to the tuple_utils file. As I have mentioned in an earlier PR, an std::tuple of device_vectors is better suited sometimes because they do not mangle the template parameter pack.